domain/storageprovisioning/service/filesystem.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+
+	"github.com/juju/juju/core/changestream"
+	"github.com/juju/juju/core/machine"
+	"github.com/juju/juju/core/watcher"
+	"github.com/juju/juju/core/watcher/eventsource"
+	"github.com/juju/juju/domain/life"
+	"github.com/juju/juju/domain/storageprovisioning"
+	"github.com/juju/juju/internal/errors"
+)
+
+// FilesystemState defines the interface required for performing filesystem
+// provisioning operations in the model.
+type FilesystemState interface {
+	// GetFilesystemAttachmentIDs returns the
+	// [storageprovisioning.FilesystemAttachmentID] information for each
+	// filesystem attachment uuid supplied. If a uuid does not exist or isn't
+	// attached to either a machine or a unit then it will not exist in the
+	// result.
+	GetFilesystemAttachmentIDs(ctx context.Context, uuids []string) (map[string]storageprovisioning.FilesystemAttachmentID, error)
+
+	// GetFilesystemAttachmentLifeForNetNode returns a mapping of filesystem
+	// attachment uuids to the current life value for each machine provisioned
+	// filesystem attachment that is to be provisioned by the machine owning the
+	// supplied net node.
+	GetFilesystemAttachmentLifeForNetNode(ctx context.Context, netNodeUUID string) (map[string]life.Life, error)
+
+	// GetFilesystemLifeForNetNode returns a mapping of filesystem ids to current
+	// life value for each machine provisioned filesystem that is to be
+	// provisioned by the machine owning the supplied net node.
+	GetFilesystemLifeForNetNode(ctx context.Context, netNodeUUID string) (map[string]life.Life, error)
+
+	// InitialWatchStatementMachineProvisionedFilesystems returns both the
```

Comment:
Sentence doesn't make sense. It says "returns both" but then only lists one thing. The following sentence isn't a complete sentence.
Same elsewhere.